atf_core/src/atf_core/recorder.py

```diff
@@ -92,21 +103,33 @@ def __init__(self, test):
                 client = SimpleActionClient(action, roslib.message.get_message_class(topic_type))
 
                 # wait for action server
-                client.wait_for_server()
+                if wait_timeout == None:
+                    client.wait_for_server()
+                else:
+                    client.wait_for_server(rospy.Duration(wait_timeout))
```

Comment:
why is this if-else case only done here?

Reply:
done in https://github.com/floweisshardt/atf/pull/186

Reply:
cherry-picked relevant changes from #186